app/main.py

```diff
@@ -37,4 +39,27 @@ def read_root(db: db_dependency):
 
 @app.post("/friend")
 def add_friend(friend: NewFriend, db: db_dependency):
-    return friend
+    db_friend = models.Friend(name=friend.name, location=friend.location, friends=friend.friends)
+    db.add(db_friend)
+    db.commit()
+    return 'Friend Added'
+
+@app.delete('/FRIENDS/{friend_id}')
+def remove_friend(friend_id: int, db: db_dependency):
+    db_query = select(models.Friend).where(models.Friend.id == friend_id)
+    for x in db.scalars(db_query):
+        db.delete(x)
+    db.commit()
+    return "Friend Removed"
+
+@app.put('/FRIENDS/{update}')
+def update_friend(friend: NewFriend, friend_id: int, db: db_dependency):
+    new_name = friend.name
+    new_location = friend.location
+    new_friends = friend.friends
+    db_query = select(models.Friend).where(models.Friend.id == friend_id)
```

Comment:
I wonder if we can use the `update` function here to write this in a more idiomatic way?

Reply:
Have you got any documentation on the .update() cause I can only see update as a parameter on the FastAPI notes?

Reply:
Our database is an array instead of a dictionary, would it work better as a dictionary?

Reply:
I got it working :)

Reply:
Could you show me how you got it? I keep trying to use .update() on either db or models.Friend but its saying it doesnt have that function